core/sql/optimizer/BindItemExpr.cpp

```diff
@@ -6173,7 +6173,35 @@ ItemExpr *Assign::bindNode(BindWA *bindWA)
     } // QSTUFF
 
   } // isUserSpecified
+  NABuiltInTypeEnum sourceType =  child(1)->castToItemExpr()->getValueId().getType().getTypeQualifier() ;
+  NABuiltInTypeEnum targetType =  child(0)->castToItemExpr()->getValueId().getType().getTypeQualifier() ;
+  if ((sourceType == NA_CHARACTER_TYPE) && (targetType == NA_LOB_TYPE))
+    {
+      ValueId vid1 = child(1)->castToItemExpr()->getValueId();  
+      // Add a stringToLob node
+      ItemExpr *newChild =  new (bindWA->wHeap()) LOBinsert( vid1.getItemExpr(), NULL, LOBoper::STRING_, FALSE);    
+      newChild->bindNode(bindWA);
+      if (bindWA->errStatus())
+	return boundExpr; 
+      setChild(1, newChild);
+    }
+  /* if ((sourceType == NA_UNKNOWN_TYPE) && (targetType = NA_LOB_TYPE))
```

Comment:
I think there was a CQD added awhile back that caused params to be treated as strings. Perhaps this commented code was intended to handle the case where the CQD is set to 'OFF'?